src/main/java/com/fauna/codec/json/QueryTagsDeserializer.java

```diff
@@ -0,0 +1,38 @@
+package com.fauna.codec.json;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+public class QueryTagsDeserializer extends JsonDeserializer<Map<String, String>> {
+    @Override
+    public Map<String,String> deserialize(JsonParser jp, DeserializationContext deserializationContext) throws IOException, JacksonException {
+        // We can improve performance by building this from tokens instead.
+        switch (jp.currentToken()) {
+            case VALUE_NULL:
+                return null;
+            case VALUE_STRING:
+                var raw = jp.getValueAsString();
```

Comment:
Possibly it's because much of my Java usage pre-dates this language feature, but every time I see `var` being used I want to replace it with the actual type. I'd kinda argue that it's worth it in public code like a driver to be as explicit as possible.

Reply:
This seems like a styling choice, but I don't really have a strong opinion. I use `var` because it improves readability to me, but I might be on an island here. Maybe the var name isn't clear enough.